src/browser/atom-application.coffee

```diff
@@ -254,8 +254,8 @@ class AtomApplication
       win = BrowserWindow.fromWebContents(event.sender)
       @applicationMenu.update(win, template, keystrokesByCommand)
 
-    ipc.on 'run-package-specs', (event, specDirectory) =>
-      @runSpecs({resourcePath: global.devResourcePath, specDirectory: specDirectory, exitWhenDone: false})
+    ipc.on 'run-package-specs', (event, specDirectory, senderDevMode) =>
+      @runSpecs({resourcePath: global.devResourcePath, specDirectory: specDirectory, exitWhenDone: false, specRequesterDevMode: senderDevMode})
```

Comment:
this extra parameter `senderDevMode` -> `specRequesterDevMode` is necessary because the spec window must request the `open` with the right `devMode`, otherwise a new window may be created.